rectools/visuals/visual_app.py

```diff
@@ -0,0 +1,273 @@
+        considered as requests. For i2i case - specific items. Their names must be specified as keys
+        of the dict and ids as values of the dict.
+    is_u2i : bool, optional
+        User-to-item recommendation case (opposed to item-to-item), by default True
+    interactions : tp.Optional[pd.DataFrame], optional
+        Table with interactions history for users. Only needed for u2i case. Supposed to be in form
+        of pandas DataFrames with columns:
+            - `Columns.User` - user id
+            - `Columns.Item` - item id
+        The original order of the rows will be preserved. Keep in mind to sort the rows correctly
+        before visualizing. The most intuitive wy is to sort by date in descending order. If user
+        has too many interactions the lest ones may not be displayed.
+        By default None.
+    auto_display : bool, optional
+        Display widgets right after initialization, by default True
+    formatters : tp.Optional[tp.Dict[str, tp.Callable]], optional
+        Formatter functions to apply to columns elements in the sections of interactions and recos.
+        Keys of the dict must be columns names (item_data, interactions and recos columns can be
+        specified here). Values bust be functions that will be applied to corresponding columns
+        elements. The result of each function must be a unicode string that represents html code.
+        Formatters can be used to format text, create links and display images with html.
+        By default None.
+    rows_limit : int, optional
+        Maximum number of rows to display in the sections of interactions and recos, by default 20
+
+    Examples
+    --------
+    >>> recos = {
+    ...     "model1": pd.DataFrame({Columns.User: [1, 2], Columns.Item: [3, 4], Columns.Score: [0.99, 0.9]}),
+    ...     "model2": pd.DataFrame({Columns.User: [1, 2], Columns.Item: [5, 6], Columns.Rank: [1, 1]})
+    ... }
+    >>>
+    >>> item_data = pd.DataFrame({
+    ...     Columns.Item: [3, 4, 5, 6, 7, 8],
+    ...     "feature_1": ["one", "two", "three", "five", "one", "two"]
+    ... })
+    >>>
+    >>> interactions = pd.DataFrame({Columns.User: [1, 1, 2], Columns.Item: [3, 7, 8]})
+    >>> requests_dict = {"user_one": 1}
+    >>> formatters = {"item_id": lambda x: f"<b>{x}</b>"}
+    >>>
+    >>> widgets = VisualApp(
+    ...     recos=recos,
+    ...     item_data=item_data,
+    ...     interactions=interactions,
+    ...     requests_dict=requests_dict,
+    ...     formatters=formatters,
+    ...     auto_display=False
+    ... )
+    """
+
+    def __init__(
+        self,
+        recos: tp.Dict[tp.Hashable, pd.DataFrame],
+        item_data: pd.DataFrame,
+        requests_dict: tp.Dict[tp.Hashable, tp.Hashable],
+        is_u2i: bool = True,
+        interactions: tp.Optional[pd.DataFrame] = None,
+        auto_display: bool = True,
+        formatters: tp.Optional[tp.Dict[str, tp.Callable]] = None,
+        rows_limit: int = 20,
```

Comment:
Don't you think 20 is too many?

Reply:
It's ok, I'm using it at 20 and it works fine. If user has 20 rows in recos df, he knows why he needs all of them :)

Reply:
ok